samples/DistributedSort/DistributedSort.fs

```diff
@@ -207,27 +213,34 @@ let main orgargs =
                                 PeerRcvdSpeedLimit = rcvdSpeedLimit )
         let finalPartitions = 256
         if nParallel > 0 then 
-            startDSet.NumParallelExecution <- nParallel    
-        if nump > 0 then 
+            startDSet.NumParallelExecution <- nParallel
+            startDSet.NumPartitions <- startDSet.Cluster.NumNodes*startDSet.NumParallelExecution
+        if nump > 0 then
             startDSet.NumPartitions <- nump
-        //let numPartitions = startDSet.NumPartitions
-        let numPartitions = startDSet.Cluster.NumNodes
-        let partitionSizeFunc( parti ) = 
+            startDSet.NumParallelExecution <- nump / startDSet.Cluster.NumNodes
+        let numPartitions = startDSet.NumPartitions
+        let partitionSizeFunc (numPartitions) ( parti ) = 
             let ba = int (num / int64 numPartitions) 
             if parti < int ( num % int64 numPartitions ) then ba + 1 else ba
         let initVectorFunc (parti, serial) =
-            let key = Array.zeroCreate<byte> nDim
-            key.[0] <- byte(serial)
-            key
+            allArray.[serial % 256]
+//            let key = Array.zeroCreate<byte> nDim
+//            key.[0] <- byte(serial)
+//            key
         let partitionFunc (kv:byte[]) = 
             int kv.[0]
-        let dkvGen = startDSet |> DSet.init initVectorFunc partitionSizeFunc 
+        let dkvGen = startDSet.InitN(initVectorFunc, partitionSizeFunc)
+        //let dkvGen = startDSet |> DSet.init initVectorFunc partitionSizeFunc 
         //let dkvGenRAM = dkvGen |> DSet.cacheInRAM
         let repartParam = DParam ( PreGroupByReserialization = 0, NumPartitions = finalPartitions )
         let dkvRepart = dkvGen.RepartitionP(repartParam, partitionFunc)
         let t3 = DateTime.UtcNow
         //let verifyCount = dkvGen |> DSet.fold (fun cnt x -> cnt + 1) (fun a b -> a + b) 0
+        let foldFunc = (fun cnt x ->
+            cnt + 1
+        )
```

Comment:
It seems like foldFunc is dead. If you'd like to keep it around, you could comment it out and add a comment explaining why it could be useful.